### DISCUSSION TITLE
What is the recommended MAX7219 LIB?

### DISCUSSION THREAD

**Discussion by GM-Script-Writer-62850** [micropython/micropython#14451] [Libraries & Drivers] [answered]
Just using it to control 7 segment displays, I do not need anything fancy, just something fast/efficient
* possibly daisy chained MAX7219

I have written code to control 7 segment panels directly via GPIO pins, i am good with something minimal

Maybe a good guide of how to use SPI to write my own using a [datasheet](https://www.analog.com/media/en/technical-documentation/data-sheets/MAX7219-MAX7221.pdf)
  * knowing how it works and making my wheel can be fun and good exp to do it for something else

**Comment by scruss** [comment, marked as answer]
Tried any of the ones from [Awesome MicroPython: LED Segment](https://awesome-micropython.com/#led-segment)?

**Reply by GM-Script-Writer-62850** [reply]
Thanks, i have found one of them with google before now (there are 2 for the 7219 written for 7 segment panels)

**Reply by GM-Script-Writer-62850** [reply]
I forked and revised the max7219_8digit lib: https://github.com/GM-Script-Writer-62850/max7219_8digit/

**Reply by scruss** [reply]
Neat! Don't forget to add your fork to Awesome MicroPython

**Reply by GM-Script-Writer-62850** [reply]
https://github.com/mcauser/awesome-micropython/issues/157

i think i ran out of thing to add to the lib, guess i could add documentation for some internal functions like `set_register` maybe i could go with, 'This is a internal function, you can use it if you want... if you are doing this do you really need this library?'
